ikarus/utils/modalanalysis/modalanalysis.hh

```diff
@@ -0,0 +1,228 @@
+    massAssembler_->bind(req_, Ikarus::AffordanceCollections::dynamics, Ikarus::DBCOption::Reduced);
+    lumpedMassAssembler_ = makeAssemblerManipulator(*massAssembler_);
+  }
+
+  /**
+   * \brief Binds a lumping scheme for the mass matrix. The Lumping scheme should have an operator () and modify the
+   * matrix mat. For details see already existing lumping schemes at \file ikarus/utils/modalanalysis/lumpingschemes.hh.
+   * It resets already bound matrix manipulation functions.
+   *
+   * \tparam LumpingScheme The type of the lumping scheme, for example one found at \file
+   * ikarus/utils/modalanalysis/lumpingschemes.hh.s
+   * \param ls the instantiated LumpingScheme object (pass either by value or by template definition).
+   */
+  template <typename LumpingScheme>
+  void bindLumpingScheme(LumpingScheme ls = {}) {
+    lumpedMassAssembler_->unbindAllMatrixFunctions();
+    lumpedMassAssembler_->bind(ls);
+  }
+
+  /**
+   * \brief Unbinds a former bound lumpingscheme.
+   */
+  void unBindLumpingScheme() { lumpedMassAssembler_->unbindAllMatrixFunctions(); }
+
+  /**
+   * \brief Starts the computation of the eigenvalue solver
+   *
+   * \param tolerance given tolerance for iterative eigenvalue solving (default: 1e-10)
+   * \param maxit givenn maximum iterations for eigenvalue solving (default: 1000)
+   * \return true solving was successful
+   * \return false solving was not successful
+   */
+  bool compute(ScalarType tolerance = 1e-10, Eigen::Index maxit = 1000) {
+    solver_.emplace(stiffAssembler_, lumpedMassAssembler_);
+    return solver_->compute(tolerance, maxit);
+  }
+
+  /**
+   * \brief Returns the angular frequencies as \f$ \omega  = \sqrt{\lambda} \f$, with \f$ \lambda \f$: eigenvalues from
+   * the eigenvalue solver
+   */
+  Eigen::VectorXd angularFrequencies() {
+    assertCompute();
+    return squaredAngularFrequencies().cwiseSqrt().eval();
+  }
+
+  /**
+   * \brief Returns the angular frequencies as \f$ f  = \dfrac{\omega}{2\pi} \f$, with \f$ \omega \f$: angular frequency
+   */
+  Eigen::VectorXd naturalFrequencies() {
+    assertCompute();
+    return angularFrequencies() / (2 * std::numbers::pi);
+  }
+
+  /**
+   * \brief Returns the angular frequencies as \f$ \omega^2  = \lambda \f$, with \f$ \lambda \f$: eigenvalues from
+   * the eigenvalue solver
+   */
+  const Eigen::VectorXd& squaredAngularFrequencies() const {
+    assertCompute();
+    return solver_->eigenvalues();
+  }
+
+  /**
+   * \brief Returns the eigenmodes (eigenvectors) of the general eigenvalue problem
+   *
+   * \return auto matrix with the eigevectors as columns
+   */
+  const Eigen::MatrixXd& eigenmodes() const { return solver_->eigenvectors(); }
+
+  /**
+   * \brief Plots the spectrum of the specified result (defaults to angular frequency) using matplot++.
+   *
+   * \param resultType specified result type of the modal analysis
+   * \param normalizeModeNumber
```

Comment:
describe `normalizeModeNumber`? To which range is the normalization done? [0,1]?

Reply:
Maybe this function can move to a specific example at some point